src/contextify.cc

```diff
@@ -180,170 +182,201 @@ class ContextWrap : public ObjectWrap {
         return true;
     }
 
-    static NAN_PROPERTY_GETTER(GlobalPropertyGetter) {
-        NanScope();
-        Local<Object> data = args.Data()->ToObject();
+    static void GlobalPropertyGetter(Local<String> property, const PropertyCallbackInfo<Value>& info) {
+        Local<Object> data = info.Data()->ToObject();
         ContextifyContext* ctx = ObjectWrap::Unwrap<ContextWrap>(data)->ctx;
-        if (!ctx)
-            NanReturnUndefined();
-        Local<Value> rv = NanNew(ctx->sandbox)->GetRealNamedProperty(property);
-        if (rv.IsEmpty()) {
-            rv = NanNew(ctx->proxyGlobal)->GetRealNamedProperty(property);
+
+        if (!ctx) {
+            info.GetReturnValue().SetUndefined();
+            return;
         }
-        NanReturnValue(rv);
+
+        Local<Value> rv = Nan::New(ctx->sandbox)->GetRealNamedProperty(property);
+
+//        if (rv.IsEmpty()) {
+//            rv = Nan::New(ctx->proxyGlobal)->GetRealNamedProperty(property);
+//        }
+
+        info.GetReturnValue().Set(rv);
     }
 
-    static NAN_PROPERTY_SETTER(GlobalPropertySetter) {
-        NanScope();
-        Local<Object> data = args.Data()->ToObject();
+    static void GlobalPropertySetter(Local<String> property, Local<Value> value, const PropertyCallbackInfo<Value>& info) {
```

Comment:
Because it is more surface area to maintain and make bugs in. The current 
policy is to have nothing unnecessary, save for some extremely convenient and 
well-defined things, or that which is (still) grandfathered in. Every part of 
the NAN API is essential. The day an Isolate argument appears in V8 master is 
the day it can be wrapped.

On Tuesday 03 November 2015 02:40:21 Ido Ben-Yair wrote:
> >      }
> > 
> > -    static NAN_PROPERTY_SETTER(GlobalPropertySetter) {
> > -        NanScope();
> > -        Local<Object> data = args.Data()->ToObject();
> > +    static void GlobalPropertySetter(Local<String> property, Local<Value>
> > value, const PropertyCallbackInfo<Value>& info) {
> v8::Context::Scope is just a RAII wrapper around Enter() and Exit(), why not
> simply alias it under NAN for now, in case it changes in the future (like
> gaining an Isolate* param)?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/brianmcd/contextify/pull/190/files#r43733995